experimental/dds/tree/src/test/Checkout.tests.ts

```diff
@@ -200,6 +198,19 @@ export function checkoutTests(
 				rangeEndpointFailure: 'Malformed',
 				status: 'Malformed',
 			});
+
+			expect(() => checkout.closeEdit()).throws('Locally constructed edits must be well-formed and valid');
+			expect(events.length).to.equal(2);
```

Comment:
the pre-existing test makes me wonder whether this is actually the correct change, rather than better documenting expectations around applyChanges etc (or some other holistic API design).

With the change, caller may end up emitting duplicate events.

This is probably less of a big deal than accidentally sending 0 events, though. Curious to hear reviewer's thoughts.

Reply:
Good point. I made a suggestion in Checkout which I think would help avoid both the double logging and the no-logging issues.